src/Node.hs

```diff
@@ -8,33 +8,53 @@
 {-# LANGUAGE KindSignatures #-}
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE RecursiveDo #-}
 
 module Node where
 
+import Control.Monad.Fix (MonadFix)
 import qualified Node.Internal as LL
-import Node.Internal (NodeId, ChannelIn, ChannelOut)
+import Node.Internal (ChannelIn(..), ChannelOut(..))
+import Data.String (fromString, IsString)
 import Data.Binary     as Bin
 import Data.Binary.Put as Bin
 import Data.Binary.Get as Bin
 import qualified Data.ByteString.Lazy as LBS
 import qualified Data.ByteString      as BS
 import qualified Data.ByteString.Builder.Extra as BS
+import Data.Map.Strict (Map)
+import qualified Data.Map.Strict as M
 import qualified Network.Transport.Abstract as NT
 import System.Random (RandomGen)
 import Mockable.Class
 import Mockable.Concurrent
 import Mockable.Channel
 import Mockable.SharedAtomic
 import Mockable.Exception
+import GHC.Generics (Generic)
 
 data Node (m :: * -> *) = Node {
        nodeLL      :: LL.Node m,
        nodeWorkers :: [ThreadId m]
      }
 
+type NodeId = LL.NodeId
+
+nodeId :: Node m -> NodeId
+nodeId = LL.NodeId . NT.address . LL.nodeEndPoint . nodeLL
+
 type Worker m = SendActions m -> m ()
 
-type MessageName = BS.ByteString
+newtype MessageName = MessageName BS.ByteString
+deriving instance Eq MessageName
+deriving instance Ord MessageName
```

Comment:
I just prefer standalone deriving syntax. I could do it in the typical way instead if you like.